Try to import from collections.abc for Python 3.3+

ixjlyons opened the pull request:
Fixes #885

ixjlyons commented on pyqtgraph/graphicsItems/ImageItem.py:
```diff
@@ -2,13 +2,16 @@
+try:
+    from collections.abc import Callable
+except ImportError:
+    from collections import Callable
```
I think so. It shouldn't emit a warning for 3.3+. It'll successfully import from `collections.abc` and not try from `collections` as it does now. Python 2.7 will fail first then get it from `collections`.

ksunden replied:
try/except is how numpy/h5py do it (mostly, a few instances of sys.version checks, but that one had other imports that were version dependent as well) 

(Also looked at matplotlib, but mpl master is py36+ only, and thus does not need any checks)

j9ac9k replied:
@ixjlyons this method would still emit a deprication warning no?

Currently (not this fork)this is the warning I get:

```
pyqtgraph/pgcollections.py:329
  /Users/ognyan/Developer/pyqtgraph/pyqtgraph/pgcollections.py:329: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    class ProtectedList(collections.Sequence):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

If we are to do the try/except method, would it not make more sense to try to import form `collections` first, and if that fails to import from `collections.abc`?  Doing it in that order should prevent any deprecation warning.

ksunden commented on pyqtgraph/pgcollections.py:
```diff
@@ -10,15 +10,21 @@
+    from collections.abc import Sequence
+except ImportError:
+    from collections import Sequence
+
```
AFAIK pyqtgraph no longer supports py26, should we combine the Sequence and OrderedDict imports, also OrderedDict import is not caught

ksunden replied:
Nevermind, OrderedDict is a concrete class not an ABC

j9ac9k commented:
I'm good with merging this, but there is now a conflict w/ `ImageItem.py` .  @ixjlyons  can you take care of that?

ixjlyons commented:
I'm not actually sure why git couldn't figure out that merge. Anyway, rebased so it should be good now. I also added a code comment so it's more likely to be noticed that the try/except block is a candidate for removal if/when dropping Python 2.7 support.

j9ac9k commented:
merged, thanks for writing this up!
